hydroflow/src/builder/build/pull_filter.rs

```diff
@@ -0,0 +1,50 @@
+use super::{PullBuild, PullBuildBase};
+
+use crate::scheduled::handoff::HandoffList;
+
+pub struct FilterPullBuild<Prev, Func>
```

Comment:
One thing I'm wondering about looking at these is, how difficult will it be for someone to implement a new operator at this level (is it even possible given type visibility? Hard for me to judge with Rust)? Do we have "escape hatch" operators that let you do ~anything? Is it possible to implement new operators just in terms of other operators (zero-cost, ideally)?

Reply:
This is a good point that I forgot about

Reply:
It is possible since I didn't actually make anything crate-private but it is not very easy or enjoyable.

I have thought a bit about macro-ing up this code in the context of reducing bloat but it would be nice for users too. But I'm not sure how possible it actually is.